mesonbuild/interpreter/interpreter.py

```diff
@@ -300,7 +300,7 @@ def __init__(
         self.configure_file_outputs: T.Dict[str, int] = {}
         # Passed from the outside, only used in subprojects.
         if default_project_options:
-            self.default_project_options = default_project_options.copy()
+            self.default_project_options = default_project_options if isinstance(default_project_options, str) else default_project_options.copy()
```

Comment:
Should this instead use listify? That handles both types, ensures that `default_project_options` is a list and also does the required shallow copy.

Reply:
Ok, this is an effect of removing `_override_options_convertor`. It's not clear why you are doing that, but in general there's brokenness due to treating options as `T.List[str]` even though they can be any of `str`, `T.List[str]` or `T.Dict[str, T.Union[str, int, bool, T.List[str]]]`; I don't understand the new code enough to understand why you didn't want to use the dictionary (which is what `_override_options_convertor` returned) as the canonical format.